src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -598,10 +587,44 @@ private static object CastOrDefault(object value, Type type)
         private object GetDefaultValue(AvaloniaProperty property)
         {
             if (property.Inherits && InheritanceParent is AvaloniaObject aobj)
-                return aobj.GetValue(property);
+                return aobj.GetValueOrDefault(property);
             return ((IStyledPropertyAccessor) property).GetDefaultValue(GetType());
         }
 
+        /// <summary>
+        /// Gets the value or default value for a property.
+        /// </summary>
+        /// <param name="property">The property.</param>
+        /// <returns>The default value.</returns>
+        private object GetValueOrDefault(AvaloniaProperty property)
```

Comment:
We should probably somehow mark methods that can be only used after access is verified. `Unsafe` prefix/postfix is an option.

Reply:
how about UnChecked as unsafe has meaning already

Reply:
+